src/sqlfluff/dialects/dialect_tsql.py

```diff
@@ -5140,3 +5143,66 @@ class CreateExternalDataSourceStatementSegment(BaseSegment):
             ),
         ),
     )
+
+
+class OpenJsonWithClauseSegment(BaseSegment):
+    """A `WITH` clause of an `OPENJSON()` table-valued function.
+
+    https://learn.microsoft.com/en-us/sql/t-sql/functions/openjson-transact-sql?view=sql-server-ver16#with_clause
+    """
+
+    type = "openjson_with_clause"
+
+    match_grammar = Sequence(
+        "WITH",
+        Bracketed(
+            Delimited(
+                AnyNumberOf(
```

Comment:
I don't think you want this any number of, as it could open things up to parse that are non-delimited